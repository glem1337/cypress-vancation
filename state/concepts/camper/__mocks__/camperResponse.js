export default {
  data: {
    id: 'a18f7338-c2b3-4cee-be4d-579c0a726c1a',
    type: 'camper',
    attributes: {
      vehicle_type_name: 'Modern Van',
    },
  },
};
