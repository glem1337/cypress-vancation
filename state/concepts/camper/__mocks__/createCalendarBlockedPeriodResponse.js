export default {
  data: {
    data: {
      id: '4b2ccd0b-448d-4dad-888f-e8bc468de6cb',
      type: 'blocked_period',
      attributes: {
        start_date: '2021-08-06',
        end_date: '2021-08-06',
        camper_id: '7384c6df-8492-49ca-814d-5ad49ed35666',
      },
    },
  },
};
