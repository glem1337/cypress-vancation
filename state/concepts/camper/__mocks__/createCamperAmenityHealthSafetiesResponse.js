export default {
  data: {
    data: [
      {
        id: '68ba4122-0843-4f53-8cbc-673b40dd0c0e',
        type: 'amenity_health_safety_item',
        attributes: {
          active: true,
        },
        relationships: {
          health_safety: {
            data: {
              id: '96b7ccbb-270e-48da-a321-f3b7827a29a5',
              type: 'health_safety',
            },
          },
        },
      },
      {
        id: '18950e99-dee7-482c-ba69-19ed02bfba01',
        type: 'amenity_health_safety_item',
        attributes: {
          active: true,
        },
        relationships: {
          health_safety: {
            data: {
              id: '96b7ccbb-270e-48da-a321-f3b7827a29a5',
              type: 'health_safety',
            },
          },
        },
      },
    ],
  },
};
