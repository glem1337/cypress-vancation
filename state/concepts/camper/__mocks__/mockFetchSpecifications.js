// eslint-disable-next-line import/prefer-default-export
export const mockedFetchSpecifications = {
  data: [
    {
      id: '92940491-75c6-434c-bf97-52ce18550b27',
      type: 'vehicle_type',
      attributes: {
        name: 'Hai Sporer',
        icon_url: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/92940491-75c6-434c-bf97-52ce18550b27/icon/55d0125c47edaa9edca5adb33aa24957.svg',
      },
      relationships: {
        vehicle_makes: {
          data: [
            {
              id: '0913145e-46b1-4fff-971c-a4e4a51eef67',
              type: 'vehicle_make',
            },
          ],
        },
      },
    },
  ],
  included: [
    {
      id: '1dfee368-b441-4286-9ef8-0c3ea80d97ed',
      type: 'vehicle_model',
      attributes: {
        name: 'Winford Erdman',
        builder_name: 'Zenia Beier',
        inside_heigh: 'Belinda Connelly',
        length: '34.78',
        year: 1156,
      },
    },
    {
      id: '4dd50b17-a49e-4683-83b2-14dbc5082259',
      type: 'vehicle_model',
      attributes: {
        name: 'Hester Dietrich',
        builder_name: 'Kizzie Christiansen',
        inside_heigh: 'Royce Reichel',
        length: '88.25',
        year: 1422,
      },
    },
    {
      id: '0913145e-46b1-4fff-971c-a4e4a51eef67',
      type: 'vehicle_make',
      attributes: { name: 'Lenny Abbott' },
      relationships: {
        vehicle_models: {
          data: [
            {
              id: '1dfee368-b441-4286-9ef8-0c3ea80d97ed',
              type: 'vehicle_model',
            },
            {
              id: '4dd50b17-a49e-4683-83b2-14dbc5082259',
              type: 'vehicle_model',
            },
          ],
        },
      },
    },
  ],
};
