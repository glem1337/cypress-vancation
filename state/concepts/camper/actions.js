import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';

import {
  CREATE_NAME_AND_DESCRIPTION,
  FETCH_CAMPER,
  CREATE_BATCH_CAMPER_PHOTOS,
  CREATE_CAMPER_PHOTO,
  DELETE_CAMPER_PHOTO,
  CREATE_CAMPER,
  FETCH_SPECIFICATIONS,
  CREATE_CAMPER_POLICIES,
  UPDATE_CAMPER_POLICIES,
  CREATE_TRIP_FEES,
  CREATE_CAMPER_PRICING,
  DELETE_CUSTOM_TRIP_FEE,
  CREATE_DELIVERY,
  UPDATE_DELIVERY,
  FETCH_CAMPER_AMENITIES,
  CHECK_PREVIOUS_STEP_COMPLETENESS,
  SET_ESTIMATE_EARNING_DATA,
  CLEAR_ESTIMATE_EARNING_DATA,
  SET_ESTIMATE_EARNING_STATE,
  CREATE_CAMPER_CALENDAR_EXPORT_LINK,
  CREATE_CAMPER_CALENDAR_IMPORT,
  SET_LEAVE_PAGE_METHOD,
  CREATE_CAMPER_AMENITIES,
  CREATE_CUSTOM_DISCOUNT_PERIOD,
  CREATE_CUSTOM_MIN_NIGHT_STAY_PERIOD,
  CREATE_CUSTOM_NIGHT_RATE_PERIOD,
  CREATE_BLOCKED_PERIODS,
  DELETE_BLOCKED_PERIODS,
  DELETE_CAMPER_EXTERNAL_CALENDAR,
  UPDATE_CAMPER_CALENDAR_IMPORT,
  FETCH_OWNER_CAMPERS,
  SET_OWNER_CAMPER_IDS,
  SET_OWNER_CAMPERS_PAGE,
  SET_OWNER_TOTAL,
  SET_OWNER_CAMPERS_STATUS_FILTER,
  SET_FIRST_PORTION_CAMPER_IDS,
  UPDATE_DEFAULT_NIGHT_RATE,
  UPDATE_BASE_DISCOUNTS,
  UPDATE_MIN_NIGHT_STAY,
  UPDATE_CAMPER_SPECIFICATIONS,
  UPDATE_CALENDAR_AVAILABILITY,
  UPDATE_PREPARATION_TIME,
  UPDATE_PICKUP_DROP_OFF_TIME,
  UPDATE_NAME_AND_DESCRIPTION,
  CREATE_INSURANCE_INFO,
  FETCH_CAMPER_CALENDAR,
  CREATE_CAMPER_AMENITY_HEALTH_SAFETIES,
  UPDATE_INSURANCE_INFO,
  UPDATE_TRIP_FEES,
  UPDATE_CAMPER_STATUS,
  DELETE_OWNER_CAMPER_ID,
  DELETE_FIRST_PORTION_CAMPER_ID,
  SET_OWNER_CAMPERS_SEARCH,
  CREATE_CAMPER_RESTRICTIONS,
  DELETE_CAMPER_RESTRICTIONS,
  CREATE_CAMPER_DOCUMENTS,
  DELETE_CAMPER_DOCUMENT,
  DELETE_CAMPER_QUESTION,
  FETCH_VEHICLE_TYPES,
  CREATE_CAMPER_TRAVEL_ACCESSORIES,
  DELETE_CAMPER_CUSTOM_TRAVEL_ACCESSORIES,
  FETCH_CAMPER_TRAVEL_ASSERTIONS,
  FETCH_CAMPER_POLICIES,
  FETCH_CAMPER_FACILITIES,
  FETCH_CAMPER_TRAVEL_EXTENTION,
  FETCH_CAMPER_OWNER,
  FETCH_CAMPER_PRICING_AND_FEES,
  SET_CAMPER_PRICING_AND_FEES_ID,
} from './types';

export const createNameAndDescription = makeFormSubmitAction(CREATE_NAME_AND_DESCRIPTION);

export const updateNameAndDescription = makeFormSubmitAction(UPDATE_NAME_AND_DESCRIPTION);

export const fetchCamper = (id, inclusions) => ({
  type: FETCH_CAMPER,
  id,
  inclusions,
});

export const createBatchCamperPhotos = ({
  camperId, photos, mainPhoto, order, redirectRoute,
}) => ({
  type: CREATE_BATCH_CAMPER_PHOTOS,
  photos,
  camperId,
  mainPhoto,
  order,
  redirectRoute,
});

export const createCamperPhoto = ({ camperId, photo }) => ({
  type: CREATE_CAMPER_PHOTO,
  photo,
  camperId,
});

export const deleteCamperPhoto = (photoId) => ({
  type: DELETE_CAMPER_PHOTO,
  photoId,
});

export const createCamperAction = makeFormSubmitAction(CREATE_CAMPER);

export const fetchSpecificationsAction = () => ({
  type: FETCH_SPECIFICATIONS,
});

export const createCamperPolicies = makeFormSubmitAction(
  CREATE_CAMPER_POLICIES,
);

export const updateCamperPolicies = makeFormSubmitAction(
  UPDATE_CAMPER_POLICIES,
);

export const createTripFees = makeFormSubmitAction(CREATE_TRIP_FEES);

export const updateTripFees = makeFormSubmitAction(UPDATE_TRIP_FEES);

export const createCamperDeliveryAction = makeFormSubmitAction(CREATE_DELIVERY);

export const updateCamperDeliveryAction = makeFormSubmitAction(UPDATE_DELIVERY);

export const createCamperPricing = makeFormSubmitAction(
  CREATE_CAMPER_PRICING,
);

export const deleteCustomTripFee = ({ feeId, camperId }) => ({
  type: DELETE_CUSTOM_TRIP_FEE,
  feeId,
  camperId,
});

export const checkPreviousStepCompleteness = ({ camperId, key, ctx }) => ({
  type: CHECK_PREVIOUS_STEP_COMPLETENESS,
  key,
  camperId,
  ctx,
});

export const setEstimateEarningData = ({ estimatedEarning, iconUrl, name }) => ({
  type: SET_ESTIMATE_EARNING_DATA,
  estimatedEarning,
  iconUrl,
  name,
});

export const clearEstimateEarningData = () => ({
  type: CLEAR_ESTIMATE_EARNING_DATA,
});

export const setEstimateEarningState = (state) => ({
  type: SET_ESTIMATE_EARNING_STATE,
  state,
});

export const createCamperCalendarExportLink = (camperId) => ({
  type: CREATE_CAMPER_CALENDAR_EXPORT_LINK,
  camperId,
});

export const createCamperCalendarImport = makeFormSubmitAction(
  CREATE_CAMPER_CALENDAR_IMPORT,
);

export const fetchCamperAmenities = (inclusions) => ({
  type: FETCH_CAMPER_AMENITIES,
  inclusions,
});

export const setLeavePageMethod = (leavePageMethod) => ({
  type: SET_LEAVE_PAGE_METHOD,
  leavePageMethod,
});

export const createCamperAmenities = makeFormSubmitAction(
  CREATE_CAMPER_AMENITIES,
);

export const createCustomDiscountPeriod = makeFormSubmitAction(CREATE_CUSTOM_DISCOUNT_PERIOD);

export const createCustomMinNightStayPeriod = makeFormSubmitAction(
  CREATE_CUSTOM_MIN_NIGHT_STAY_PERIOD,
);

export const createCustomNightRatePeriod = makeFormSubmitAction(
  CREATE_CUSTOM_NIGHT_RATE_PERIOD,
);

export const createBlockedPeriods = ({ camperId, startDate, endDate }) => ({
  type: CREATE_BLOCKED_PERIODS,
  camperId,
  startDate,
  endDate,
});

export const deleteBlockedPeriods = ({ camperId, startDate, endDate }) => ({
  type: DELETE_BLOCKED_PERIODS,
  camperId,
  startDate,
  endDate,
});

export const deleteCamperExternalCalendar = ({ calendarId, camperId }) => ({
  type: DELETE_CAMPER_EXTERNAL_CALENDAR,
  calendarId,
  camperId,
});

export const createInsuranceInfo = (camperId) => ({
  type: CREATE_INSURANCE_INFO,
  camperId,
});

export const updateInsuranceInfo = makeFormSubmitAction(UPDATE_INSURANCE_INFO);

export const fetchCamperCalendar = ({ camperId, startDate, endDate }) => ({
  type: FETCH_CAMPER_CALENDAR,
  camperId,
  startDate,
  endDate,
});

export const updateCamperCalendarImport = makeFormSubmitAction(
  UPDATE_CAMPER_CALENDAR_IMPORT,
);

export const fetchOwnerCampers = () => ({
  type: FETCH_OWNER_CAMPERS,
});

export const setOwnerCamperIds = camperIds => ({
  type: SET_OWNER_CAMPER_IDS,
  camperIds,
});

export const deleteOwnerCamperId = (camperId) => ({
  type: DELETE_OWNER_CAMPER_ID,
  camperId,
});

export const setFirstPortionCamperIds = camperIds => ({
  type: SET_FIRST_PORTION_CAMPER_IDS,
  camperIds,
});

export const deleteFirstPortionCamperId = camperId => ({
  type: DELETE_FIRST_PORTION_CAMPER_ID,
  camperId,
});

export const setOwnerCampersPage = page => ({
  type: SET_OWNER_CAMPERS_PAGE,
  page,
});

export const setOwnerTotal = total => ({
  type: SET_OWNER_TOTAL,
  total,
});

export const setOwnerCampersStatusFilter = status => ({
  type: SET_OWNER_CAMPERS_STATUS_FILTER,
  status,
});

export const setOwnerCampersSearch = search => ({
  type: SET_OWNER_CAMPERS_SEARCH,
  search,
});

export const updateDefaultNightRate = makeFormSubmitAction(UPDATE_DEFAULT_NIGHT_RATE);

export const updateBaseDiscounts = makeFormSubmitAction(UPDATE_BASE_DISCOUNTS);

export const updateMinNightStay = makeFormSubmitAction(UPDATE_MIN_NIGHT_STAY);

export const updateCamperSpecification = makeFormSubmitAction(UPDATE_CAMPER_SPECIFICATIONS);

export const updateCalendarAvailability = makeFormSubmitAction(UPDATE_CALENDAR_AVAILABILITY);

export const updateCalendarPreparationTime = makeFormSubmitAction(UPDATE_PREPARATION_TIME);

export const updatePickupDropOffTime = makeFormSubmitAction(UPDATE_PICKUP_DROP_OFF_TIME);

export const createCamperAmenityHealthSafeties = makeFormSubmitAction(
  CREATE_CAMPER_AMENITY_HEALTH_SAFETIES,
);

export const updateCamperStatus = ({ status, camperId }) => ({
  type: UPDATE_CAMPER_STATUS,
  status,
  camperId,
});

export const createCamperRestrictions = makeFormSubmitAction(CREATE_CAMPER_RESTRICTIONS);

export const deleteCamperCustomRestrictions = ({ id, camperId, customRestrictionType }) => ({
  type: DELETE_CAMPER_RESTRICTIONS,
  id,
  camperId,
  customRestrictionType,
});

export const createCamperDocuments = makeFormSubmitAction(
  CREATE_CAMPER_DOCUMENTS,
);

export const deleteCamperDocument = ({
  documentId,
  camperId,
}) => ({
  type: DELETE_CAMPER_DOCUMENT,
  documentId,
  camperId,
});

export const deleteCamperQuestion = ({
  questionId,
  camperId,
}) => ({
  type: DELETE_CAMPER_QUESTION,
  questionId,
  camperId,
});

export const fetchVehicleTypes = () => ({
  type: FETCH_VEHICLE_TYPES,
});

export const createCamperTravelAccessories = makeFormSubmitAction(
  CREATE_CAMPER_TRAVEL_ACCESSORIES,
);

export const deleteCamperCustomTravelAccessory = ({ addonId, camperId }) => ({
  type: DELETE_CAMPER_CUSTOM_TRAVEL_ACCESSORIES,
  addonId,
  camperId,
});

export const fetchCamperTravelAssertions = (id) => ({
  type: FETCH_CAMPER_TRAVEL_ASSERTIONS,
  id,
});

export const fetchCamperPolicies = (id) => ({
  type: FETCH_CAMPER_POLICIES,
  id,
});

export const fetchCamperFacilities = (id) => ({
  type: FETCH_CAMPER_FACILITIES,
  id,
});

export const fetchCamperTravelExtention = (id) => ({
  type: FETCH_CAMPER_TRAVEL_EXTENTION,
  id,
});

export const fetchCamperOwner = (id) => ({
  type: FETCH_CAMPER_OWNER,
  id,
});

export const fetchCamperPricingAndFees = ({
  camperId,
  startDate,
  endDate,
}) => ({
  type: FETCH_CAMPER_PRICING_AND_FEES,
  camperId,
  startDate,
  endDate,
});

export const setCamperPricingAndFeesId = (id) => ({
  type: SET_CAMPER_PRICING_AND_FEES_ID,
  id,
});
