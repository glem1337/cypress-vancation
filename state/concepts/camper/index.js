import camperReducer from './reducer';

export { default as camperOperations } from './operations';
export * as camperTypes from './types';

export default camperReducer;
