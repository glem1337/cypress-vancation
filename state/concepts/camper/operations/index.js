import createNameAndDescription from './createNameAndDescription';
import fetchCamper from './fetchCamper';
import createCamperPhoto from './createCamperPhoto';
import createCamperPhotosBatch from './createCamperPhotosBatch';
import deleteCamperPhoto from './deleteCamperPhoto';
import createCamperOperation from './createCamperOperation';
import fetchSpecificationsOperation from './fetchSpecificationsOperation';
import createCamperPoliciesOperation from './createCamperPolicies';
import updateCamperPolicies from './updateCamperPolicies';
import createTripFees from './createTripFees';
import createDelivery from './createDelivery';
import updateDelivery from './updateDelivery';
import createCamperPricingOperation from './createCamperPricing';
import deleteCustomTripFee from './deleteCustomTripFee';
import fetchCamperAmenities from './fetchCamperAmenities';
import checkPreviousStepCompleteness from './checkPreviousStepCompleteness';
import createCamperCalendarExportLink from './createCamperCalendarExportLink';
import createCamperCalendarImport from './createCamperCalendarImport';
import createCamperAmenities from './createCamperAmenities';
import createCustomDiscountPeriod from './createCustomDiscountPeriod';
import createCustomMinNightStayPeriod from './createCustomMinNightStayPeriod';
import createCustomNightRatePeriod from './createCustomNightRatePeriod';
import createCalendarBlockedPeriods from './createCalendarBlockedPeriods';
import deleteCalendarBlockedPeriods from './deleteCalendarBlockedPeriods';
import updateCamperCalendarImport from './updateCamperCalendarImport';
import deleteCamperExternalCalendar from './deleteCamperExternalCalendar';
import fetchOwnerCampers from './fetchOwnerCampers';
import updateDefaultNightRate from './updateDefaultNightRate';
import updateBaseDiscounts from './updateBaseDiscounts';
import updateMinNightStay from './updateMinNightStay';
import updateCamperSpecifications from './updateCamperSpecifications';
import updateCalendarAvailability from './updateCalendarAvailability';
import updatePreparationTime from './updatePreparationTime';
import updatePickupDropOffTime from './updatePickupDropOffTime';
import updateNameAndDescription from './updateNameAndDescription';
import createInsuranceInfo from './createInsuranceInfo';
import fetchCamperCalendar from './fetchCamperCalendar';
import createCamperAmenityHealthSafeties from './createCamperAmenityHealthSafeties';
import updateInsuranceInfo from './updateInsuranceInfo';
import updateTripFees from './updateTripFees';
import updateCamperStatus from './updateCamperStatus';
import createCamperRestrictions from './createCamperRestrictions';
import deleteCamperCustomRestrictions from './deleteCamperCustomRestrictions';
import createCamperDocuments from './createCamperDocuments';
import deleteCamperDocument from './deleteCamperDocument';
import deleteCamperQuestion from './deleteCamperQuestion';
import fetchVehicleTypesOperation from './fetchVehicleTypesOperation';
import createCamperTravelAccessories from './createCamperTravelAccessories';
import deleteCamperCustomTravelAccessory from './deleteCamperCustomTravelAccessory';
import fetchCamperTravelAssertions from './fetchCamperTravelAssertions';
import fetchCamperPolicies from './fetchCamperPolicies';
import fetchCamperFacilities from './fetchCamperFacilities';
import fetchCamperTravelExtention from './fetchCamperTravelExtention';
import fetchCamperOwner from './fetchCamperOwner';
import fetchCamperPricingAndFees from './fetchCamperPricingAndFees';

export default [
  createNameAndDescription,
  fetchCamper,
  createCamperPhotosBatch,
  createCamperPhoto,
  deleteCamperPhoto,
  createCamperOperation,
  fetchSpecificationsOperation,
  createCamperPoliciesOperation,
  updateCamperPolicies,
  createTripFees,
  createCamperPricingOperation,
  deleteCustomTripFee,
  createDelivery,
  updateDelivery,
  fetchCamperAmenities,
  checkPreviousStepCompleteness,
  createCamperCalendarExportLink,
  createCamperCalendarImport,
  createCamperAmenities,
  createCustomDiscountPeriod,
  createCustomMinNightStayPeriod,
  createCustomNightRatePeriod,
  createCalendarBlockedPeriods,
  deleteCalendarBlockedPeriods,
  updateCamperCalendarImport,
  deleteCamperExternalCalendar,
  fetchOwnerCampers,
  updateDefaultNightRate,
  updateBaseDiscounts,
  updateMinNightStay,
  updateCalendarAvailability,
  updatePreparationTime,
  updatePickupDropOffTime,
  updateNameAndDescription,
  createInsuranceInfo,
  fetchCamperCalendar,
  updateCamperSpecifications,
  createCamperAmenityHealthSafeties,
  updateInsuranceInfo,
  updateTripFees,
  updateCamperStatus,
  createCamperRestrictions,
  deleteCamperCustomRestrictions,
  createCamperDocuments,
  deleteCamperDocument,
  deleteCamperQuestion,
  fetchVehicleTypesOperation,
  createCamperTravelAccessories,
  deleteCamperCustomTravelAccessory,
  fetchCamperTravelAssertions,
  fetchCamperPolicies,
  fetchCamperFacilities,
  fetchCamperTravelExtention,
  fetchCamperOwner,
  fetchCamperPricingAndFees,
];
