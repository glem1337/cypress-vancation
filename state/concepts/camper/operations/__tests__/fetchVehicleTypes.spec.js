import normalize from 'json-api-normalizer';

import mockHttpClient from 'utils/testHelpers/mockHttpClient';
import { mockedFetchSpecifications } from 'state/concepts/camper/__mocks__/mockFetchSpecifications';
import { fetchVehicleTypesEndpoint } from 'state/concepts/camper/endpoints';
import { dataApiFailure, dataApiRequest, dataApiSuccess } from 'state/data/actions';
import showErrorNotifications from 'utils/showErrorNotifications';

import fetchVehicleTypesOperations from '../fetchVehicleTypesOperation';

jest.mock('utils/showErrorNotifications', () => jest.fn());

describe('fetchVehicleTypesOperations', () => {
  let dispatch;
  const { url, endpoint } = fetchVehicleTypesEndpoint;

  const beforeFunction = httpClient => () => {
    dispatch = jest.fn();
    fetchVehicleTypesOperations.process({ httpClient, getState: jest.fn }, dispatch, jest.fn());
  };

  it('snapshot', () => {
    expect(fetchVehicleTypesOperations).toMatchSnapshot();
  });

  describe('success', () => {
    const httpClient = mockHttpClient({ method: 'get', response: mockedFetchSpecifications });

    beforeEach(beforeFunction(httpClient));

    it('calls right endpoint', () => {
      expect(httpClient.spy).toHaveBeenCalledWith(url);
    });

    it('dispatches actions', () => {
      expect(dispatch).toHaveBeenCalledTimes(2);

      expect(dispatch).toHaveBeenNthCalledWith(1, dataApiRequest({ endpoint }));

      expect(dispatch).toHaveBeenNthCalledWith(2,
        dataApiSuccess({
          response: normalize(mockedFetchSpecifications.data),
          endpoint,
        }));
    });
  });

  describe('failure', () => {
    const error = new Error('test');

    const httpClient = mockHttpClient({ method: 'get', reject: true, response: error });

    beforeEach(beforeFunction(httpClient));

    it('dispatches actions', () => {
      expect(dispatch).toHaveBeenCalledTimes(2);

      expect(dispatch).toHaveBeenNthCalledWith(1, dataApiRequest({ endpoint }));

      expect(dispatch).toHaveBeenNthCalledWith(2, dataApiFailure({ endpoint }));
    });

    it('shows error', () => {
      expect(showErrorNotifications).toHaveBeenCalledWith(error, dispatch);
    });
  });
});
