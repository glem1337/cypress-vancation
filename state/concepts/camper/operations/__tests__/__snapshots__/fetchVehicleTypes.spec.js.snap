// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchVehicleTypesOperations snapshot 1`] = `
Object {
  "cancelType": undefined,
  "debounce": 0,
  "latest": true,
  "name": undefined,
  "process": [Function],
  "processOptions": Object {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "camper/FETCH_VEHICLE_TYPES",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
